packages/backend/test/peripherals/database/BalanceRepository.test.ts

```diff
@@ -194,4 +197,52 @@ describe(BalanceRepository.name, () => {
 
     expect(result).toEqual([])
   })
+
+  it(BalanceRepository.prototype.getLatestPerHolder.name, async () => {
+    const blockNumberRepository = new BlockNumberRepository(knex, Logger.SILENT)
+
+    const additionalRecords = [
+      {
+        blockNumber: START_BLOCK_NUMBER,
+        holderAddress: MOCK_HOLDER,
+        assetId: AssetId('token-a'),
+        balance: MOCK_BALANCE,
+      },
+    ]
+
+    repository.addOrUpdateMany(additionalRecords)
+
+    DATA.map((d, index) =>
+      blockNumberRepository.add({
+        timestamp: START.add(index, 'hours'),
+        blockNumber: d.blockNumber,
+      }),
+    )
```

Comment:
@sin6pi7 @antooni This causes nondeterministic behavior in tests because the calls are not awaited.

Reply:
Gonna update that